react-shopping-cart/src/index.jsx

```diff
@@ -1,36 +1,52 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
-import { createStore } from 'redux';
+import { createStore, applyMiddleware } from 'redux';
 import { Provider } from 'react-redux';
 import { persistStore } from 'redux-persist';
 import { PersistGate } from 'redux-persist/integration/react';
+import thunk from 'redux-thunk';
 import { ThemeProvider } from 'styled-components';
 import rootReducer from 'reducers';
-import { ProductList, NotFound } from 'pages';
+import { ProductList, ProductDetail, NotFound, ShoppingBasket } from 'pages';
 import { GlobalStyle, theme } from 'styles';
 import PATH from 'constants/path';
 
-const store = createStore(rootReducer);
+const store = createStore(rootReducer, applyMiddleware(thunk));
 const persistor = persistStore(store);
 
+if (process.env.NODE_ENV === 'development') {
+  if (window.location.pathname === process.env.PUBLIC_URL) {
+    window.location.pathname = `${process.env.PUBLIC_URL}/`;
+  }
```

Comment:
URL 끝에 /가 없으면 MSW로 mocking한 api가 호출되지 않는 에러를 만났습니다.

/가 없을 때 발생하는 에러는 아래와 같습니다.

> Cannot intercept requests on this page because it's outside of the worker's scope ("[http://localhost:3000/react-shopping-cart/](http://localhost:3000/react-shopping-cart/)"). If you wish to mock API requests on this page, you must resolve this scope issue.

scope와 관련된 이슈인 거 같습니다. URL 끝에 /를 붙여주면 해당 이슈가 해결되고 mocking한 api가 정상적으로 호출돼서 해당 if 문을 사용했습니다!

혹시 해당 이슈와 관련해서 적절한 해결 방법이 있을까요..?

Reply:
공식문서에도 해당 부분에대해서 나와있네요!
https://mswjs.io/docs/getting-started/integrate/browser#using-homepage-property-in-packagejson